### PR TITLE
Use AddPermissionProvider & AddNavigationProvider in OpenID

### PR DESCRIPTION
**Pull request by hishamco**

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
Can you please do the same thing everywhere so we don't have 1 PR per file?

https://github.com/search?q=repo%3AOrchardCMS%2FOrchardCore+ServiceDescriptor.Scoped%3C&type=code

**Comment by hishamco** [comment]
The aim is not to replace the `ServiceDescriptor` in this PR but to use the extension methods when it's possible

**Comment by hishamco** [comment]
These are the only not updated in the code base

**Comment by MikeAlhayek** [comment]
To make this PR more useful, I suggest to replace all these 

https://github.com/search?q=repo%3AOrchardCMS%2FOrchardCore+services.TryAddEnumerable&type=code

this way we don't see many PR like this doing the same thing just for another service.

**Comment by hishamco** [comment]
Just one thing I'm afraid of is to break existing behavior coz TryAddXXX is not similar to AddXXX

File: src/OrchardCore/OrchardCore.Infrastructure/Entities/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -10,7 +10,8 @@ public static class ServiceCollectionExtensions
     public static IServiceCollection AddIdGeneration(this IServiceCollection services)
     {
         services.TryAddSingleton<IIdGenerator, DefaultIdGenerator>();
-        services.TryAddEnumerable(ServiceDescriptor.Singleton<IGlobalMethodProvider, IdGeneratorMethod>());
+        services.TryAddSingleton<IGlobalMethodProvider, IdGeneratorMethod>();
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.Infrastructure/Entities/ServiceCollectionExtensions.cs:13]: I don't think try add is needed

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**